packages/flutter/lib/src/widgets/modal_barrier.dart

```diff
@@ -12,6 +15,105 @@ import 'gesture_detector.dart';
 import 'navigator.dart';
 import 'transitions.dart';
 
+// Recognizes tap down by any pointer button unconditionally. When it receives a
+// PointerDownEvent, it immediately claims victor of arena and calls
+// `onAnyTapDown` without any checks.
+//
+// It is used by ModalBarrier to detect any taps on the overlay.
+class _AnyTapGestureRecognizer extends PrimaryPointerGestureRecognizer {
+  _AnyTapGestureRecognizer({
+    Object debugOwner,
+    this.onAnyTapDown,
+  }) : super(debugOwner: debugOwner);
+
+  VoidCallback onAnyTapDown;
+
+  bool _sentTapDown = false;
+
+  @override
+  void addAllowedPointer(PointerDownEvent event) {
+    super.addAllowedPointer(event);
+    resolve(GestureDisposition.accepted);
+  }
+
+  @override
+  void handlePrimaryPointer(PointerEvent event) {
+    if (!_sentTapDown) {
+      if (onAnyTapDown != null)
+        onAnyTapDown();
+      _sentTapDown = true;
+    }
+  }
+
+  @override
+  void didStopTrackingLastPointer(int pointer) {
+    super.didStopTrackingLastPointer(pointer);
+    _sentTapDown = false;
+  }
+
+  @override
+  String get debugDescription => 'any tap';
+}
+
+class _ModalBarrierSemanticsDelegate extends SemanticsGestureDelegate {
+  const _ModalBarrierSemanticsDelegate({this.onAnyTapDown});
```

Comment:
nit: the formatting between this line and line 70 is inconsistent.

Reply:
Fixed